server/user.js

```diff
@@ -0,0 +1,188 @@
+ * @param {string} userPath Directory of user indormation file.
+ * @param {!user.userInfo} userInfo User Information
+ * @return {user.Error|boolean}
+ */
+user.signUp = function(userPath, userInfo) {
+  var checkDuplicate = {
+    duplicated: false,
+    elements: []
+  };
+  var lines = fs.readFileSync(userPath + user.userInfoFile).toString()
+    .split('\n');
+  for (var i = 0; i < lines.length; i++) {
+    var parts = lines[i].split(/[\t\s]+/);
+    if (parts.length == 0) {
+      continue;
+    }
+    if (parts[0] == userInfo.email || parts[1] == userInfo.username) {
+      checkDuplicate.duplicated = true;
+      if (parts[0] == userInfo.email) {
+        checkDuplicate.elements.push('email: ' + userInfo.email);
+      }
+      if (parts[1] == userInfo.username) {
+        checkDuplicate.elements.push('username: ' + userInfo.username);
+      }
+      break;
+    }
+  }
+  if (checkDuplicate.duplicated) {
+    var errorMessage = checkDuplicate.elements.join(' and ') + ' exist';
+    errorMessage += checkDuplicate.elements.length == 1 ? 's' : '';
+    return {
+      error: {
+        type: 'sign-up',
+        message: errorMessage
+      }
+    };
+  } else {
+    var infoLine = userInfo.email + ' ' + userInfo.username + ' ' +
+      userInfo.password + ' ' + userInfo.confirmed + '\n';
+    fs.appendFile(userPath + user.userInfoFile, infoLine, function(err) {
+      if (err) {
+        log.serverLog(err);
+        return;
+      }
+      log.serverLog('user information saved');
+    });
+    var folder = userPath + userInfo.username + '/';
+    if (!fs.existsSync(folder)) {
+      fs.mkdirSync(folder);
+    }
+    return true;
+  }
+};
+
+/**
+ * Checks the user information for signing in.
+ * @param {string} userPath Directory of user information file.
+ * @param {!user.userInfo} userInfo User Information
+ * @return {user.Error|boolean}
+ */
+user.signIn = function(userPath, userInfo) {
+  var lines = fs.readFileSync(userPath + user.userInfoFile).toString()
+    .split('\n');
+  for (var i = 0; i < lines.length; i++) {
+    var parts = lines[i].split(/[\t\s]+/);
+    if (parts.length == 0) {
+      continue;
+    }
+    if (parts[1] == userInfo.username && parts[2] == userInfo.password) {
+      return true;
+    }
+  }
+  return {
+    error: {
+      type: 'sign-in',
+      message: 'invalid username or password'
+    }
+  };
+};
+
+/**
+ * Gets session ID from database. Regenerate if it is expired.
+ * @param {!mongodb.Db} db Session database.
+ * @param {string} username Username of the POST query.
+ * @param {!Object} res HTTP response.
+ * @param {function()} callback Callback function.
+ */
+user.findUserInfo = function(db, username, res, callback) {
+  var cursor = db.collection('session').find({username: username});
+  var result = [];
+  cursor.each(function(err, doc) {
+    assert.equal(err, null, '');
+    if (doc != null) {
+      result.push(doc);
+    } else {
+      callback();
```

Comment:
Actually after a closer look I found this a bit strange. 

1) below you overwrite the callback function. Thus your original passed in callback (db.close) may never get called.
2) you use cursor.each, but for each callback you are sending back response. But cursor each iterates results and can run multiple times, in which case you call multiple callbacks. However in this case your cursor should have only one selected item, so it may not go wrong. But the way the code executes does not quite follow the workflow expected.